test/test_cli.go

```diff
@@ -1254,3 +1254,52 @@ func (s *CLISuite) TestSlugReleaseGarbageCollection(t *c.C) {
 		s.assertURI(t, slugs[i], http.StatusOK)
 	}
 }
+
+func (s *CLISuite) TestDockerPush(t *c.C) {
+	// build image with ENV and CMD
+	repo := "cli-test-push"
+	s.buildDockerImage(t, repo,
+		`ENV FOO=BAR`,
+		`CMD ["/bin/pingserv"]`,
+	)
+
+	// create app
+	client := s.controllerClient(t)
+	app := &ct.App{Name: "cli-test-docker-push"}
+	t.Assert(client.CreateApp(app), c.IsNil)
+
+	// TODO: remove this once CI configures DockerPushURL correctly
```

Comment:
Well, this PR changes CI so that it sets `DockerPushURL`, but it won't be in use until we merge and restart CI, so I have added a hack to get the tests to pass with a plan to drop the commit just before merge.

Not ideal but after a couple of tries, I couldn't get CI to support both master and this branch.

Also note I have moved the two hacks into a single place: d537450